tests/console/yast2_lan_device_settings.pm

```diff
@@ -36,49 +36,57 @@ sub run {
     script_run('ip a');
     script_run('ls -alF /etc/sysconfig/network/');
     save_screenshot;
-
-    my $opened = open_yast2_lan();
-    wait_still_screen;
-    if ($opened eq "Controlled by network manager") {
-        return;
+    unless (check_var("ARCH", "s390x")) {
+        my $opened = open_yast2_lan();
+        wait_still_screen;
+        if ($opened eq "Controlled by network manager") {
+            return;
+        }
+
+        send_key "alt-i";    # open edit device dialog
+        assert_screen 'edit-network-card';
+        send_key "alt-t";    # select static IP address option
+        send_key "tab";
+        type_string $static_ip;
+        send_key 'alt-o';
+        send_key_until_needlematch 'hostname-localhost-removed', 'backspace';
+        type_string $static_hostname;
+        send_key "alt-n";    # next
+        wait_still_screen;
+        assert_screen qw(yast_lan_duplicate_ip static-ip-address-set);
+        if (match_has_tag 'yast_lan_duplicate_ip') {
+            send_key "alt-n";
+            send_key "alt-c";
+            close_yast2_lan();
+            record_soft_failure("Duplicate IP, $static_ip is currently unavailable. Skipping static IP assignment");
+        } elsif (match_has_tag 'static-ip-address-set') {
+            close_yast2_lan();
+
+            # verify that static IP has been set
+            assert_script_run "ip a | grep $static_ip";
+            # verify that hostname for static ip is recorded in /etc/hosts
+            $is_set_in_etc_host->($static_hostname) or die qq{Static hostname "$static_hostname" was not written to /etc/hosts file!\n};
+
+            open_yast2_lan();
+            for (1 .. 2) { send_key "tab" }    # move to device list
```

Comment:
@sofiasyria Can you run VR against non-s390x to observe the behaviour! i have the impression that this could be moved later in the code. i think you can move some part later in the test instead of open and close again the yast2_lan. But i would like to see a VR first.

Reply:
ok.

Reply:
But Sofia did not change that part of code, so it should work as before. Not sure that we need to change something here in scope of the PR.

Reply:
About open and close I agree that it's not optimal, but this is not my code, it's green just because of the new indentation. I just added a condition that includes old code.